secure/storage/github/src/lib.rs

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use thiserror::Error;
+
+/// Request timeout for github operations
+const ACCEPT_HEADER: &str = "Accept";
+const ACCEPT_VALUE: &str = "Accept: application/vnd.github.v3+json";
+const TIMEOUT: u64 = 10_000;
+const URL: &str = "https://api.github.com";
+
+#[derive(Debug, Error, PartialEq)]
+pub enum Error {
+    #[error("Http error: {1}")]
+    HttpError(u16, String),
+    #[error("Internal error: {0}")]
+    InternalError(String),
+    #[error("Missing field {0}")]
+    MissingField(String),
+    #[error("404: Not Found: {0}")]
+    NotFound(String),
+    #[error("Serialization error: {0}")]
+    SerializationError(String),
+}
+
+impl From<std::io::Error> for Error {
+    fn from(error: std::io::Error) -> Self {
+        Self::SerializationError(format!("{}", error))
+    }
+}
+
+impl From<ureq::Response> for Error {
+    fn from(resp: ureq::Response) -> Self {
+        Error::HttpError(resp.status(), resp.status_line().into())
+    }
+}
+
+impl From<serde_json::Error> for Error {
+    fn from(error: serde_json::Error) -> Self {
+        Self::SerializationError(format!("{}", error))
+    }
+}
+
+/// Client provides a client around the restful interface to GitHub API version 3. Learn more
+/// here: https://developer.github.com/v3
+///
+/// This is not intended for securely storing private data, though perhaps it could with a private
+/// repository. The tooling is intended to be used to exchange data in an authenticated fashion
+/// across multiple peers.
+pub struct Client {
+    repository: String,
+    owner: String,
+    token: String,
+}
+
+impl Client {
+    pub fn new(owner: String, repository: String, token: String) -> Self {
+        Self {
+            repository,
+            owner,
+            token,
+        }
+    }
+
+    /// Delete a file from a GitHub repository
+    pub fn delete(&self, path: &str) -> Result<(), Error> {
+        let hash = self
+            .get_sha(path)?
+            .ok_or_else(|| Error::InternalError(format!("Path does not exist: {}", path)))?;
```

Comment:
nit: seems like this should be a NotFound error, right? i.e., get_sha() returned Ok(None), which internally means a 404?

Reply:
good eye, I cleaned up this mess